codice/ddf
# DDF-2386 Removed default resource download limit, allowing unlimited data option

#### What does this PR do?
The resource management application previously enforced an arbitrary default data usage limit of  750MB for all users. This PR removes the default limit and instead allows _unlimited_ data downloads for all users. 

To apply the data download limits, the administrator can set the following special values: 
` -1` Unlimited data usage
`0` Restrict all data usage, no further download capability
Other values greater than 0 will apply normally as the data usage limit in MB/GB.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining 
@rzwiefel 
@jrnorth 
@lessarderic 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith 
@kcwire 

#### How should this be tested?
1. Build and install DDF. 
2. From the Admin console, activate the Resource Management application.
3. Configure the Resource Management application to "`Monitor local sources`"
4. Ingest data.
5. Download (repeatedly, at least 2x) the ingested resource. 
6. Verify in the Admin console under DDF Resource Management Data Consumption, that the user's data usage has increased and `Data Remaining` is Unlimited. 
7. Continue modifying the data limits for single user and all users to:
  - `0` -> Verify no downloads permitted.
  - `750` -> Verify data used and data remaining values are updated appropriately
  - `-1` -> Verify data is unrestricted

#### What are the relevant tickets?
As an administrator, I need a configurable resource download data limit
https://codice.atlassian.net/browse/DDF-2386

Relates To: 
ResourceUsagePlugin does not count first download against user's limit
https://codice.atlassian.net/browse/DDF-2368

#### Checklist:
